core/lib/vm_interface/src/executor.rs

```diff
@@ -20,6 +20,7 @@ pub trait BatchExecutorFactory<S: Send + 'static>: 'static + Send + fmt::Debug {
         storage: S,
         l1_batch_params: L1BatchEnv,
         system_env: SystemEnv,
+        pubdata_params: PubdataParams,
```

Comment:
Dumb question: Do `PubdataParams` not fit into either `L1BatchEnv` or `SystemEnv`? Or would they fit *now*, but cease to fit in the future when custom `PubdataBuilder`s would be implemented? In general, what's the relationship between `PubdataParams` and a `PubdataBuilder`? Is it possible to create a builder from params, or not necessarily?

Reply:
So `PubdataParams` is data that we persist for batches in DB and that we use for building `PubdataBuilder`s. `PubdataParams` is not customizable for now as it requires additional effort and it's not yet clear how to make it so and wire it altogether with node framework. I'm leaving it as an intermediate step until we really need custom `PubdataBuilder`s. Recent changes were to prepare `VmInterface` for custom impls but not the node itself.